deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/security/UserDataHelper.java

```diff
@@ -82,8 +84,15 @@ public String getUsername(String token) throws Exception {
         Map<String, String> params = ImmutableMap.of("access_token", token);
         String jsonPayload = httpClient.get(userDataUrl, params, null, 3);
         JsonObject jsonObject = new JsonParser().parse(jsonPayload).getAsJsonObject();
-        jsonObject = jsonObject.getAsJsonObject("user");
-        String user = jsonObject.get("username").getAsString();
+        final String user;
+        if (oauthProvider.equals("googlev2")){
```

Comment:
I would agree - let us keep the oauthProvider directive, but treat Pinterest Provider a special case. Any other providers will fall back to default behavior